dd_bot:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Effort Controllers ---------------------------------------
  left_wheel_hinge_effort_controller:
    type: effort_controllers/JointEffortController
    joint: left_wheel_hinge
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_wheel_hinge_effort_controller:
    type: effort_controllers/JointEffortController
    joint: right_wheel_hinge
    pid: {p: 100.0, i: 0.01, d: 10.0}

  # Position Controllers ---------------------------------------
  left_wheel_hinge_position_controller:
    type: effort_controllers/JointPositionController
    joint: left_wheel_hinge
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_wheel_hinge_position_controller:
    type: effort_controllers/JointPositionController
    joint: right_wheel_hinge
    pid: {p: 100.0, i: 0.01, d: 10.0}


  # Velocity Controllers ---------------------------------------
  left_wheel_hinge_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: left_wheel_hinge
    pid: {p: 1.0, i: 0.00, d: 0.1}
  right_wheel_hinge_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: right_wheel_hinge
    pid: {p: 1.0, i: 0.00, d: 0.1}

dd_mpc:
  Q_flat: [1.0, 0.0, 0.0, 
           0.0, 1.0, 0.0, 
           0.0, 0.0, 0.1]

  P_flat: [1.0, 0.0, 0.0, 
           0.0, 1.0, 0.0, 
           0.0, 0.0, 0.1]

  R_flat: [0.1, 0.0, 
           0.0, 0.1]

  Q_dim: 3
  P_dim: 3
  R_dim: 2

  horizon_length: 20
  time_step: 0.2

LC_traj:

  a1: 5.0
  a2: 5.0
  w1: 0.3
  w2: 0.2
  del: 0


wheel_velocity_controller:

  control_rate_hz: 50.0

  kp: 1.0
  ki: 0.0
  kd: 0.2

dd_bot_geometry:

  wheel_dist: 0.355
  wheel_radius: 0.1
